src/testUtils/TestRig.react.js

```diff
@@ -1,46 +1,80 @@
-import React from 'react'
-
-export var TestRigComponent = React.createClass({
-  contextTypes: {
-    triggerComplete: React.PropTypes.func.isRequired,
-    finish: React.PropTypes.func
-  },
-  getInitialState: function() {
-    return { keyVal : Math.random()};
-  },
-  componentDidUpdate: function(prevProps, prevState) {
-    this.context.finish()
-  },
-  triggerFinalRender: function() {
-    this.setState({keyVal: Math.random()})
-  },
+import React from 'react/addons'
+var TestUtils = React.addons.TestUtils
+
+export class TestRigComponent extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { keyVal : Math.random(), testComplete: false }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if(this.state.testComplete=== true)
+      this.state.expectationCallback()
+  }
+
+  triggerFinalRender() {
+    this.setState({testComplete: true})
+    //this.setState({keyVal: Math.random()}) -- probably don't need this anymore, but holding onto just in case it's useful again
+  }
+
   render() {
     return <div key={this.state.keyval}>{this.props.children}</div>
```

Comment:
@BJK I know this wasn't changed in this PR, but why is there a `key` on a component not created in a loop?